server/src/main/java/org/elasticsearch/cluster/InternalClusterInfoService.java

```diff
@@ -262,10 +261,10 @@ public void onResponse(IndicesStatsResponse indicesStatsResponse) {
                     reservedSpaceBuilders.forEach((nodeAndPath, builder) -> rsrvdSpace.put(nodeAndPath, builder.build()));
 
                     indicesStatsSummary = new IndicesStatsSummary(
-                        Collections.unmodifiableMap(shardSizeByIdentifierBuilder),
-                        Collections.unmodifiableMap(shardDataSetSizeBuilder),
-                        Collections.unmodifiableMap(dataPathByShardRoutingBuilder),
-                        Collections.unmodifiableMap(rsrvdSpace)
+                        Map.copyOf(shardSizeByIdentifierBuilder),
+                        Map.copyOf(shardDataSetSizeBuilder),
+                        Map.copyOf(dataPathByShardRoutingBuilder),
+                        Map.copyOf(rsrvdSpace)
```

Comment:
Since type of the map is critical, would it be better to move this inside of the constructor so that it is impossible to create the object with wrong (slower) map implementation?